docs/resources/scope.md

```diff
@@ -0,0 +1,97 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "squaredup_scope Resource - terraform-provider-squaredup"
+subcategory: ""
+description: |-
+  SquaredUp Scope
+---
+
+# squaredup_scope (Resource)
+
+SquaredUp Scope
+
+## Example Usage
+
+```terraform
+data "squaredup_datasources" "sample_data" {
+  data_source_name = "Sample Data"
+}
+
+resource "squaredup_datasource" "sample_data_source" {
+  display_name     = "Sample Data"
+  data_source_name = data.squaredup_datasources.sample_data.plugins[0].display_name
+}
+
+resource "squaredup_workspace" "application_workspace" {
+  display_name      = "Application Team"
+  description       = "Workspace with Dashboards for Application Team"
+  datasources_links = [squaredup_datasource.sample_data_source.id]
+}
+
+resource "squaredup_scope" "advanced_scope" {
+  scope_type     = "advanced"
+  display_name   = "Advanced Scope"
+  workspace_id   = squaredup_workspace.application_workspace.id
+  advanced_query = "g.V().has('__configId', '${squaredup_datasource.sample_data_source.id}').has('sourceId', 'sample-server-2')" //any gremlin query
+}
+
+data "squaredup_nodes" "acommon_node" {
+  depends_on     = [squaredup_datasource.sample_data_source]
+  data_source_id = squaredup_datasource.sample_data_source.id
+  node_name      = "account-common-lambda"
+}
+
+data "squaredup_nodes" "api_node" {
+  depends_on     = [squaredup_datasource.sample_data_source]
+  data_source_id = squaredup_datasource.sample_data_source.id
+  node_name      = "master-api-lambda"
+}
+
+resource "squaredup_scope" "fixed_scope" {
+  scope_type   = "fixed"
+  display_name = "Fixed Scope"
+  workspace_id = squaredup_workspace.application_workspace.id
+  node_ids     = [data.squaredup_nodes.acommon_node.node_properties[0].id, data.squaredup_nodes.api_node.node_properties[0].id]
```

Comment:
What's with the `node_properties[0]` - is this a Terraform thing?  How many sets of properties will a node have?  If always just the one perhaps we can hide this if not a Terraform "thing"

Reply:
Thanks @P2P-Nathan. That is correct.

Reply:
My question was more about the choosing of the slightly confusing `node_properties` variable name.  Did we choose this or is it a Terraform thing?  Where is it defined / chosen?

Reply:
No this was chosen. The reason I had used `node_properties` is because the api returns these data in `properties` object. There is also another `properties` within the `properties` object therefore I went for a slightly different name